include/zephyr/device.h

```diff
@@ -184,6 +184,44 @@ typedef int16_t device_handle_t;
 			&Z_DEVICE_STATE_NAME(Z_DEVICE_DT_DEV_ID(node_id)),     \
 			__VA_ARGS__)
 
+/**
+ * @brief Create a sub-device object from a devicetree node identifier and set
+ * it up for boot time initialization.
+ *
+ * This macro defines a @ref device that is automatically configured by the
+ * kernel during system initialization. The global device object's name as a C
+ * identifier is derived from the node's dependency ordinal. @ref device.name is
+ * set to `DEVICE_DT_NAME(node_id)`.
+ *
+ * Sub-device takes state and power management from parent device, so any
+ * operations affecting power management or state will be applied to parent
+ * device.
```

Comment:
definitely -1 to state sharing. why can't this be accomplished by sharing some code between separate parent/child PM callbacks?